app/controllers/users_controller.rb

```diff
@@ -1,49 +1,67 @@
 class UsersController < ApplicationController
 
-  before_action :require_login, except: [:new, :create]
-  before_action :validate_user, only: [:show]
+  before_action :authenticate_user, except: [:new, :create]
+  before_action :save_login_state, only: [:new]
 
   def new
-    @user = User.new(user_params[:user])
+    @new_user = User.new(user_params[:user])
   end
 
-# trip login so user is logged in as soon as they create the account
+  # trip login so user is logged in as soon as they create the account
   def create
     @user = User.create(user_params[:user])
     if @user.save
-      redirect_to login_path
       flash[:success] = "Registration successful, take your shiny new username and password for a spin"
+
+      redirect_to login_path
     else
       flash.now[:errors] = "Registration invalid, please try again."
-      render new_user_path
+
+      redirect_to register_path
     end
   end
 
   def show
-    @user_id = params[:id]
-    @user = User.find(@user_id)
+    sales_quantity
+    total_sales
+    @recent_sales = @user.recent_sales
+
+    render :show
   end
 
+  def edit
+  end
+
+  def update
+    @user.update(user_params[:user])
+    @user.save
+
+    redirect_to user_path
+  end
 
   private
 
-  # this could be added to any other controllers that also require login
-  def require_login
-    unless session[:user_id]
-      redirect_to login_path
-    end
+  def user_params
+    params.permit(user: [:username, :email, :password, :password_confirmation])
   end
 
-  def validate_user
-    if @user.id == session[:user_id]
-      render :show
-    else
-    redirect_to user_path(session[:user_id])
-    flash.now[:errors] = "Access denied"
+  def total_sales
+    @total_sales = 0
+    @user.order_items.each do |x|
+      if x.order.status != "cancelled"
+        @total_sales += x.quantity * x.product.price
+      end
     end
+    @total_sales=@total_sales/100
   end
 
-  def user_params
-    params.permit(user: [:username, :email, :password, :password_confirmation])
+  def sales_quantity
+    @quantity_sold = 0
+    sales_array = @user.order_items
+    sales_array.each do |x|
+      if x.order.status != "cancelled"
```

Comment:
I see a lot of conditionals around the order status. If you isolate these into model scopes or methods, you can simplify your controller logic.